tasks/swig.rake

```diff
@@ -11,24 +13,32 @@
 def run_swig(mod)
   swig = `which swig`.chomp
   swig = `which swig2.0`.chomp if swig.empty?
+  swiglib = `swig -swiglib`.chomp
+  abort unless $?.success?
 
   # Standard search location for headers
-  include_args = %w[-I/usr/local/include -I/usr/include]
+  include_args = "-I#{Taglib.install_dir}/include"
 
   if ENV.key?('TAGLIB_DIR')
     unless File.directory?(ENV['TAGLIB_DIR'])
       abort 'When defined, the TAGLIB_DIR environment variable must point to a valid directory.'
     end
 
     # Push it in front to get it searched first.
-    include_args.unshift("-I#{ENV['TAGLIB_DIR']}/include")
+    include_args = "-I#{ENV['TAGLIB_DIR']}/include"
   end
 
-  sh "cd ext/#{mod} && #{swig} -c++ -ruby -autorename -initname #{mod} #{include_args.join(' ')} #{mod}.i"
+  Dir.chdir("ext/#{mod}") do
+    sh "#{swig} -c++ -ruby -autorename -initname #{mod} #{include_args} #{mod}.i"
+    wrap = "#{mod}_wrap.cxx"
+    wrapdata = File.read(wrap)
+    File.write(wrap, wrapdata.gsub(%r{#{swiglib}}, '/swig'))
```

Comment:
Let's use a string as the first argument for `gsub` here, we want to replace all the strings, not do any regex matching (e.g. `.` in the path):

```suggestion
    File.write(wrap, wrapdata.gsub(swiglib, '/swig'))
```